spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java

```diff
@@ -65,13 +74,29 @@
 	 */
 	public AppDeploymentRequest(AppDefinition definition, Resource resource,
 			Map<String, String> environmentProperties) {
+		this(definition, resource, environmentProperties, null);
+	}
+
+	/**
+	 * Construct an {@code AppDeploymentRequest}.
+	 *
+	 * @param definition app definition
+	 * @param resource resource for the underlying app's artifact
+	 * @param environmentProperties map of environment properties; may be {@code null}
+	 * @param commandlineArguments set of command line arguments; may be {@code null}
+	 */
+	public AppDeploymentRequest(AppDefinition definition, Resource resource,
+			Map<String, String> environmentProperties, Set<String> commandlineArguments) {
```

Comment:
Well I picked set not to have duplicate entries, why allowing dups? Though even set really doesn't guarantee that if one entry has multiple options.